src/jit/gentree.h

```diff
@@ -3774,8 +3801,9 @@ struct GenTreeCall final : public GenTree
         // only used for CALLI unmanaged calls (CT_INDIRECT)
         GenTree* gtCallCookie;
         // gtInlineCandidateInfo is only used when inlining methods
-        InlineCandidateInfo*   gtInlineCandidateInfo;
-        void*                  gtStubCallStubAddr;              // GTF_CALL_VIRT_STUB - these are never inlined
+        InlineCandidateInfo*                  gtInlineCandidateInfo;
+        GuardedDevirtualizationCandidateInfo* gtGuardedDevirtualizationCandidateInfo;
```

Comment:
We probably need to handle the new field in `Compiler::gtCloneExpr`. If not, we would see JITStress failures.

Reply:
Thanks for pointing this out. Unfortunately we can't safely clone inline candidates in isolation like this, as they and the associated `GT_RET_EXPR`s must refer to one another. Also we can't safely share inline candidate info across multiple calls.

To put it another there are some special and unusual rules in the IR early on and any cloning of an inline candidate requires cloning of the `GT_RET_EXPR` and special post processing to fix things up (which is a good part of what the guarded devirt transformation and up-front `GT_RET_EXPR` spilling does).

So what goes on here when cloning calls is wrongish but can't easily be made right. We should I suppose null out the candidate information as that would be somewhat less wrong, but it is still wrong. And we need to clone calls so banishing cloning for inline candidates or guarded devirtualization candidates outright (like we do for `GT_RET_EXPR`s) doesn't help either.

I doubt stress mode cloning runs before we've inlined as it would already be busted (and after we get past `fgInline`  these special  complications vanish). So I further doubt these changes will pose any new problems beyond the ones we already have and are avoiding.

But I could be wrong. I'll keep an eye on cloning as I continue working on this feature, but for now don't expect to need to change this aspect.

Reply:
Thinking about this a bit more, probably the right change is to ban cloning of inline candidates via `gtCloneExpr` and introduce a separate method that can be used to clone them.

Because inline candidate calls can't ever be subtrees or live on the eval stack it's unlikely anyone would clone them inadvertently (say as part of cloning some ancestor tree), so having a deliberate API to call in this case seems like it might be a good compromise.

Will follow up on this in a subsequent change.